### ISSUE TITLE
secure_satis: true does not work with BasicAuth

### ISSUE THREAD

**Issue by NeoBlack** [terramar-labs/packages#84]
I have set secure_satis: true, the packages.json is now protected by a BasicAuth but entering the credentials does not enable the access to the files.
not in browser and also not by composer

**Comment by tyler-sommer** [comment]
If you change the username/password to something extremely basic such as `admin` and `password`, does it work? I'm wondering if there is a problem when parsing the yaml configuration. Additionally, you may need to clear your cache directory.

You could also do the classic `die()` debugging, for example in: https://github.com/terramar-labs/packages/blob/master/src/Application.php#L56 throw a `die($security['username']);` in there to verify that the configuration is being read correctly.

**Comment by NeoBlack** [comment]
Config is correct, $security contains the username and password from the config.yml.
after entering the data, the dialog comes back and ask again for the credentials.
I will try to setup the the tool on my local machine today, it's better for debugging.

**Comment by strausmann** [comment]
Hello Tyler, could you until the implementation of the feature (https://github.com/terramar-labs/packages/issues/77) add a second user?
This user can only access satis. I would like to secure satis and allow access for certain users. But would like to avoid that you log into the admin.

Many thanks,
Regards,

Bjoern

**Comment by strausmann** [comment]
Hello Tyler, 

i think I have found the mistake. The [password_hash()](https://github.com/terramar-labs/packages/blob/master/src/Application.php#L61) function use with the PASSWORD_DEFAULT algo a bcrypt-Algorithmus Password. But in the config.yml File is the passwort in plain-text saved.

Regards,
Bjoern

**Comment by NeoBlack** [comment]
good catch @strausmann so the fix is to use an encrypted password hash (which the best idea) but will the login still works? will check it in the evening.

**Comment by NeoBlack** [comment]
so, on my local system the BasicAuth for /packages.json work with a plaintext password., on my web server it does not work. on both system an encrypted password does not work.

**Comment by tyler-sommer** [comment]
The password is hashed [during application boot](https://github.com/terramar-labs/packages/blob/master/src/Application.php#L61), though I agree the password should be hashed in the config file itself. This would be considered a BC break, however, so I want to avoid it if possible on the 3.x line.

The password entered during authentication is also hashed before comparison, so you should enter the plaintext password when authenticating.

I admit, though, I'm at a loss at what could be causing your issue, especially given it working locally. What does your setup look like, in both environments-- how are they different? Could it be a webserver configuration causing the problem?

**Comment by NeoBlack** [comment]
It must not be a  breaking change, an encrypted password (and the algorithm) can be detected by the prefix. example:

plain: password
brcypt: $2a$04$ntozz44SYhGgHBUEMSvYKOvWmVqMVC9v.PVfeDuEYzr7L.BL7r7e2

check the possible prefixes:
https://en.wikipedia.org/wiki/Bcrypt#Versioning_History

sure, the local setup is different from the server, but the server "normal" debian webserver. I will attach a [PHPInfo output](https://github.com/terramar-labs/packages/files/1728917/phpinfo.pdf)

**Comment by tyler-sommer** [comment]
Excellent suggestion, @NeoBlack, I think that's a great way forward. There is already a utility in `vendor/bin/hashpass` that is shipped with nice-security. This utility could be used during a Composer post-install hook for the user.

I've opened https://github.com/terramar-labs/packages/issues/91 to track that feature.

However, this doesn't address your issues with authenticating. Is that still an issue?

**Comment by NeoBlack** [comment]
yes it is still an issue, I had not the time yet to debug it on the server or have a deeper look into the code. Will try to check it the next days.
